src/pages/components/utils/index.ts

```diff
@@ -38,3 +39,32 @@ export const mapColorByExecution = (planned: number, actual: number) => {
     return 'red'
   }
 }
+
+export type HourlyDataList = HourlyData[]
+ // Convert gapsList into HourlyDataList structure
+export const processData = (gapsList: GapsList): HourlyDataList => {
```

Comment:
having this name is a bad encapsulation - please refactor.
when someone reads the code, they should understand what this function is doing without having to read the entire implementation

Reply:
also - now i have a thought: may be to move it to `service`?